serve/pull.go

```diff
@@ -96,13 +96,31 @@ func (s *Service) pull(rw http.ResponseWriter, r *http.Request) {
 	}
 
 	clientViewURL := ""
-	// TODO enable multiple client view URLs and auto-add ASID URLs.
-	if len(acct.ClientViewURLs) > 0 {
-		clientViewURL = acct.ClientViewURLs[0]
-	}
-	if s.overridClientViewURL != "" {
-		clientViewURL = s.overridClientViewURL
+	if preq.Version >= 3 {
+		if preq.ClientViewURL == "" {
+			clientError(rw, http.StatusBadRequest, "clientViewURL not provided in request", l)
+			return
+		}
+		clientViewURL = preq.ClientViewURL
+		authorized, err := account.ClientViewURLAuthorized(s.maxASClientViewURLs, s.accountDB, accounts, acct.ID, clientViewURL)
+		if err != nil {
+			serverError(rw, err, l)
+			return
+		}
+		if !authorized {
+			clientError(rw, http.StatusForbidden, "clientViewURL is not authorized; please contact support@replicache.dev", l)
+			return
+		}
+	} else {
+		// TODO remove this block when Version 2 is deprecated.
+		if len(acct.ClientViewURLs) > 0 {
+			clientViewURL = acct.ClientViewURLs[0]
+		}
+		if s.overridClientViewURL != "" {
```

Comment:
typo? Should it not be overrid_e_ClientViewURL?

Reply:
heh, yes it should be. addressed in https://github.com/rocicorp/diff-server/pull/90